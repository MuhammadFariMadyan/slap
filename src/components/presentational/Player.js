import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import {
  MdKeyboardArrowLeft,
  MdKeyboardArrowRight,
  MdClose
} from "react-icons/md";

import MainSlide from "./MainSlide";

const Container = styled.div`
  position: fixed;
  width: 100%;
  height: 100vh;
  background: black;
  display: flex;
  align-items: center;
`;

export const ArrowLeft = styled(MdKeyboardArrowLeft)`
  color: white;
  font-size: 96px;
  padding: 4px;
  cursor: pointer;
`;

export const ArrowRight = ArrowLeft.withComponent(MdKeyboardArrowRight);

export const Close = styled(MdClose)`
  position: absolute;
  top: 32px;
  right: 32px;
  font-size: 48px;
  color: #555;
  cursor: pointer;
`;

const Player = ({
  active,
  slides,
  onClickLeft,
  onClickRight,
  onClickClose
}) => {
  const slide = slides.find((_, index) => index === active);

  return (
    <Container>
      <ArrowLeft onClick={onClickLeft} />
      <MainSlide contents={slide.contents} />
      <ArrowRight onClick={onClickRight} />
      <Close onClick={onClickClose} />
    </Container>
  );
};

Player.propTypes = {
  active: PropTypes.number,
  slides: PropTypes.array,
  onClickLeft: PropTypes.func,
  onClickRight: PropTypes.func,
  onClickClose: PropTypes.func
};

export default Player;
