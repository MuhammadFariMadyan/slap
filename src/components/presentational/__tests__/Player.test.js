import React from "react";
import { shallow } from "enzyme";

import Player, { ArrowLeft, ArrowRight, Close } from "../Player";

jest.mock("../MainSlide", () => "MainSlide");

describe("Player", () => {
  const setup = propOverrides => {
    const props = {
      active: 0,
      slides: [
        {
          id: "slide-1",
          projectID: "project-1",
          contents: [
            {
              id: "content-1",
              slideID: "slide-1",
              projectID: "project-1",
              type: "text",
              content: "Stay hungry",
              position: {
                x: 128,
                y: 96
              }
            }
          ]
        },
        {
          id: "slide-2",
          projectID: "project-1",
          contents: [
            {
              id: "content-2",
              slideID: "slide-2",
              projectID: "project-1",
              type: "text",
              content: "Stay foolish",
              position: {
                x: 128,
                y: 128
              }
            }
          ]
        }
      ],
      onClickLeft: jest.fn(),
      onClickRight: jest.fn(),
      onClickClose: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Player {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClickLeft' callback while left clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(ArrowLeft).simulate("click");

    expect(props.onClickLeft).toBeCalled();
  });

  it("should calls 'onClickRight' callback while left clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(ArrowRight).simulate("click");

    expect(props.onClickRight).toBeCalled();
  });

  it("should calls 'onClickClose' callback while close clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(Close).simulate("click");

    expect(props.onClickClose).toBeCalled();
  });
});
