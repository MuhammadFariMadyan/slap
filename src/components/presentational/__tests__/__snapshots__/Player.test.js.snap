// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should renders default 1`] = `
<styled.div>
  <Styled(Component)
    onClick={[Function]}
  />
  <MainSlide
    contents={
      Array [
        Object {
          "content": "Stay hungry",
          "id": "content-1",
          "position": Object {
            "x": 128,
            "y": 96,
          },
          "projectID": "project-1",
          "slideID": "slide-1",
          "type": "text",
        },
      ]
    }
  />
  <Styled(Component)
    onClick={[Function]}
  />
  <Styled(Component)
    onClick={[Function]}
  />
</styled.div>
`;
